src/requests/team.ts

```diff
@@ -119,3 +144,58 @@ export async function _destroyTeam(
     );
   return Boolean(response.delete_teams_by_pk?.id);
 }
+
+export async function upsertMember(
+  client: GraphQLClient,
+  args: UpsertMember,
+): Promise<boolean> {
+  const response: { insert_team_members_one: { id: number } | null } =
+    await client.request(
+      gql`
+        mutation UpsertTeamMember(
+          $role: user_roles_enum = viewer
+          $team_id: Int
+          $user_id: Int
+        ) {
+          insert_team_members_one(
+            object: { team_id: $team_id, user_id: $user_id, role: $role }
+            on_conflict: {
+              constraint: team_members_user_id_team_id_key
+              update_columns: role
```

Comment:
This is super handy - you can't allow more a user to have multiple roles within the same team. This works for the current, very simple, model but may not last forever.